src/contextbuilder.cpp

```diff
@@ -130,6 +201,20 @@ Context* ContextBuilder::deviceOpen()
 	return deviceOpen(lst.at(0).c_str());
 }
 
+M2k *ContextBuilder::m2kOpen(struct iio_context* ctx)
+{
+	auto dev = deviceOpen(ctx);
+	if (!dev) {
```

Comment:
style-wise, some prefer to not add brackets when the block is a single line;
this is for if() or for(), while() blocks;

so, you could do
``` 
if (!dev)
    return nullptr;
```

the brackets usually add a new line, which is not usually needed

Reply:
I consider using the brackets in a single line block is maintainability-wise, because adding more lines afterwards, without adding curly braces could cause a lot of logical errors.